### ISSUE TITLE
Known spammers

### ISSUE THREAD

**Issue by hello-smile6** [webdev03/scratch-spam#11]
- PowerUpProduction123 (maybe, found on griffpatch's profile)
  ```
  PowerUpProduction123 posted "Hello there ‮ <a href="/users/Griffpatch">@Griffpatch</a>!" which in
  cludes a disallowed char/string.
  ```

**Comment by webdev03** [comment]
RTL is included.

**Comment by hello-smile6** [comment]
> RTL is included.

Huh. Is LTR? Also, could you add a `start` script to `package.json`?

**Comment by webdev03** [comment]
LTR is too. I don't think the start script is necessary. I'm trying to set up building for releases.

**Comment by hello-smile6** [comment]
> LTR is too. I don't think the start script is necessary. I'm trying to set up building for releases.

Use `https://npmjs.com/package/pkg` and add support for entering the password and username from a prompt if the environment variable isn't set.

**Comment by webdev03** [comment]
I'm setting up building now.

**Comment by webdev03** [comment]
Actually, nevermind. It's too hard!

**Comment by hello-smile6** [comment]
> Actually, nevermind. It's too hard!

I'll do it

**Comment by webdev03** [comment]
I got building to work, but it can't import the `disallow.json` on my machine...

**Comment by hello-smile6** [comment]
> I got building to work, but it can't import the `disallow.json` on my machine...

Strange. Maybe `require()` it? Also, binaries will include OpenSSL in the binary, which is pretty big, for HTTPS support.

**Comment by webdev03** [comment]
Require wouldn't let us change it

**Comment by hello-smile6** [comment]
> Require wouldn't let us change it

Huh

**Comment by hello-smile6** [comment]
```
hello-smile6@localhost MINGW64 ~/Documents/scratch-spam (main)
$ pkg --compress Brotli -t node16-x64 index.js
> pkg@5.5.2
compression:  Brotli

hello-smile6@localhost MINGW64 ~/Documents/scratch-spam (main)
$ pkg --compress Brotli -t node16-win-x64 index.js
> pkg@5.5.2
compression:  Brotli

hello-smile6@localhost MINGW64 ~/Documents/scratch-spam (main)
$ ls
auth.js        index.exe*  LICENSE        package.json    README.md
disallow.json  index.js    node_modules/  pnpm-lock.yaml

hello-smile6@localhost MINGW64 ~/Documents/scratch-spam (main)
$ index.exe
bash: index.exe: command not found

hello-smile6@localhost MINGW64 ~/Documents/scratch-spam (main)
$ ./index.exe
Scratch Spam v0.1.2
What user would you like to check for spam? 9pfs
Found 28 comment thread starters!
Would you like to go to the next page? (Y/n) y
Found 56 comment thread starters!
Would you like to go to the next page? (Y/n) y
Found 72 comment thread starters!
Would you like to go to the next page? (Y/n) y
No more comments able to be found.
Starting checking...
0 bad comment(s) found!
Exiting program...

hello-smile6@localhost MINGW64 ~/Documents/scratch-spam (main)
$
```

**Comment by webdev03** [comment]
Wow

**Comment by hello-smile6** [comment]
> Wow

We just need a way for people to sign in. Also, could you please make it read users' messages and delete from there? Profiles aren't the only places spammers hit.

**Comment by hello-smile6** [comment]
@webdev03 Error:
```
node:internal/process/task_queues:96
    runMicrotasks();
    ^

Error: Failed to log in with 9pfs,{my password}
    at ScratchSession.init (C:\snapshot\scratch-spam\auth.js)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async C:\snapshot\scratch-spam\index.js

```

**Comment by webdev03** [comment]
If it's someone else's project/profile you can't do anything. I might add project support in the future :)

**Comment by hello-smile6** [comment]
> If it's someone else's project/profile you can't do anything. I might add project support in the future :)

Just read the messages! Just send a request to api.scratch.mit.edu/users/${username}/messages with the user's session ID and CSRF token.

**Comment by hello-smile6** [comment]
> If it's someone else's project/profile you can't do anything. I might add project support in the future :)
> 
> Edit: Advertisement protection could be there too.

Good idea. Then we could have a user-maintained version of `filter.scratch.mit.edu` that we actually have control over, and that we could add entries to a lot faster than the admins could. We could even make a browser extension that lets people report comments to be added to the blacklist, for cases when something is spam but they don't want to tell the site admins.

**Comment by webdev03** [comment]
I know. But studio replies are very rare and have a limit now. Projects could be supported too. I'd also like it so that you don't need to log in to an account to use Scratch Spam, so that you can verify it works.

**Comment by hello-smile6** [comment]
> I know. But studio replies are very rare and have a limit now. Projects could be supported too. I'd also like it so that you don't need to log in to an account to use Scratch Spam, so that you can verify it works.

Good idea. Also, use the `--compression Brotli` flag on `pkg`, so binaries are smaller.

**Comment by webdev03** [comment]
But it doesn't work

**Comment by hello-smile6** [comment]
> But it doesn't work
Try this.
```
npx pkg --compress Brotli -t node16 index.js
chmod u+x `find ./index.*`
chmod u-x ./index.js
```
Then, execute the `index` binary.

**Comment by webdev03** [comment]
OK, now I need to take the Ninetails builder and edit it.

**Comment by hello-smile6** [comment]
> OK, now I need to take the Ninetails builder and edit it.

Won't work. That uses electron. Use `pkg`. Do `npm i -g pkg` and try it.

**Comment by webdev03** [comment]
I made it.

**Comment by hello-smile6** [comment]
> I made it.

Oh. Could you just use `pkg`? People might want to build their own binary without installing anything.

**Comment by webdev03** [comment]
This is getting a bit offtopic.

A blacklist for users I'll add tomorrow. You can "block" people by combining it with a cron job (or just a script on replit)
